src/classiceditorui.js

```diff
@@ -77,51 +78,105 @@ export default class ClassicEditorUI extends EditorUI {
 	init( replacementElement ) {
 		const editor = this.editor;
 		const view = this.view;
+		const editingView = editor.editing.view;
+		const editable = view.editable;
+		const editingRoot = editingView.document.getRoot();
 
 		view.render();
 
-		// Set–up the sticky panel with toolbar.
-		view.stickyPanel.bind( 'isActive' ).to( this.focusTracker, 'isFocused' );
-		view.stickyPanel.limiterElement = view.element;
-
-		if ( this._toolbarConfig.viewportTopOffset ) {
-			view.stickyPanel.viewportTopOffset = this._toolbarConfig.viewportTopOffset;
-		}
-
-		// Setup the editable.
-		const editingRoot = editor.editing.view.document.getRoot();
-		view.editable.bind( 'isReadOnly' ).to( editingRoot );
-		view.editable.bind( 'isFocused' ).to( editor.editing.view.document );
-		view.editable.name = editingRoot.rootName;
-
-		this._editableElements.set( view.editable.name, view.editable.element );
-
-		this.focusTracker.add( view.editable.element );
-
-		view.toolbar.fillFromConfig( this._toolbarConfig.items, this.componentFactory );
-
-		enableToolbarKeyboardFocus( {
-			origin: editor.editing.view,
-			originFocusTracker: this.focusTracker,
-			originKeystrokeHandler: editor.keystrokes,
-			toolbar: view.toolbar
+		// The editable UI element in DOM is available for sure only after the editor UI view has been rendered.
+		// But it can be available earlier if a DOM element has been passed to BalloonEditor.create().
+		const editableElement = editable.element;
+
+		// The editable UI and editing root should share the same name. Then name is used
+		// to recognize the particular editable, for instance in ARIA attributes.
+		editable.name = editingRoot.rootName;
+
+		// Register the editable UI view in the editor. A single editor instance can aggregate multiple
+		// editable areas (roots) but the classic editor has only one.
+		this._editableElements.set( editable.name, editableElement );
+
+		// Let the global focus tracker know that the editable UI element is focusable and
+		// belongs to the editor. From now on, the focus tracker will sustain the editor focus
+		// as long as the editable is focused (e.g. the user is typing).
+		this.focusTracker.add( editableElement );
+
+		// Let the editable UI element respond to the changes in the global editor focus
+		// tracker. It has been added to the same tracker a few lines above but, in reality, there are
+		// many focusable areas in the editor, like balloons, toolbars or dropdowns and as long
+		// as they have focus, the editable should act like it is focused too (although technically
+		// it isn't), e.g. by setting the proper CSS class, visually announcing focus to the user.
+		// Doing otherwise will result in editable focus styles disappearing, once e.g. the
+		// toolbar gets focused.
```

Comment:
Not sure if we need that much docs. Especially since it is repeated in each class. I think that they say not only why but also what the code does what is not needed.

Reply:
I mean, it is a cool idea for a docs guide: what needs to be defined to create your own editor (if we do not have such guide yet, and we could link such guild in all our editors, but the fact it is that long make the code harder to read and because it is repeaded it is hard to maintain.

Reply:
Each time I do anything around here in `*EditorUI` I find myself lost. At first, I wrote the docs for myself. Then I figured that this might actually improve DX. We should definitely create a guide for that purpose ("my own editor from scratch") but I'd feel more comfortable if those comments stayed.